src/io/PipeWriter.zig

```diff
@@ -1233,11 +1279,39 @@ pub fn WindowsStreamingWriter(
                 return .{ .done = 0 };
             }
 
+            if (this.source != null and this.source.? == .sync_file) {
+                defer this.outgoing.reset();
+                var remain = StreamBuffer.writeOrFallback(&this.outgoing, buffer, comptime writeFn) catch {
+                    return .{ .err = bun.sys.Error.oom };
+                };
+                const initial_len = remain.len;
+                const fd = bun.toFD(this.source.?.sync_file.file);
+
+                while (remain.len > 0) {
```

Comment:
```suggestion
                while (true) {
```
both the condition and the one in `.result` do a 0 check which may be okay.